livetwo/src/rtspclient.rs

```diff
@@ -766,20 +766,105 @@ fn generate_digest_response(
 #[cfg(test)]
 mod tests {
     use super::*;
+    use tokio::io::{AsyncReadExt, AsyncWriteExt};
+    use tokio::net::{TcpListener, TcpStream};
 
-    #[test]
-    fn test_generate_digest_response() {
-        let username = "username";
-        let password = "password";
-        let uri = "/resource";
-        let realm = "Realm";
-        let nonce = "1234567890";
-        let method = "GET";
+    #[tokio::test]
+    async fn test_send_describe_request() {
+        let listener = TcpListener::bind("127.0.0.1:0").await.unwrap();
+        let addr = listener.local_addr().unwrap();
 
-        let expected_response = "5a8a58beeb78f36ed2c0f0d474288f3d";
+        let client = TcpStream::connect(addr).await.unwrap();
+        let (mut server, _) = listener.accept().await.unwrap();
 
-        let response = generate_digest_response(username, password, uri, realm, nonce, method);
+        let sdp_content = "v=0\r\no=- 12345 12345 IN IP4 127.0.0.1\r\ns=Test\r\nm=video 0 RTP/AVP 96\r\na=rtpmap:96 H264/90000\r\n";
+        let content_length = sdp_content.len();
 
-        assert_eq!(response, expected_response);
+        tokio::spawn(async move {
+            let mut buffer = vec![0; 4096];
+            let n = server.read(&mut buffer).await.unwrap();
+            let _ = String::from_utf8_lossy(&buffer[..n]);
+
+            let response = format!(
+            "RTSP/1.0 200 OK\r\nCSeq: 1\r\nContent-Type: application/sdp\r\nContent-Length: {}\r\n\r\n{}",
+            content_length,
+            sdp_content
+        );
+
+            server.write_all(response.as_bytes()).await.unwrap();
+            server.flush().await.unwrap();
+        });
+
+        let mut rtsp_session = RtspSession {
+            stream: client,
```

Comment:
There use type `TcpStream`, But there only use `write_all()` and `read()`

There can use Generics: `AsyncReadExt + AsyncWriteExt`. After use `tokio::io::duplex` create a `io` pair, This io working in memory